eole/transforms/tokenize.py

```diff
@@ -172,24 +172,42 @@ def _tokenize(self, tokens, side="src", is_train=False):
         # This method embeds a custom logic to correctly handle certain placeholders
         # in case the tokenizer doesn't preserve them.
         sentence = " ".join(tokens)
-        # Locate the end-of-sentence placeholders.
-        sent_list = sentence.split(self.eos_token)
+        if self.mapped_tokens is not None:
+            delim_list = [mapped_toks[0] for mapped_toks in self.mapped_tokens] + [
+                self.eos_token
+            ]
+        else:
+            delim_list = [self.eos_token]
+        pattern = f"({'|'.join(map(re.escape, delim_list))})"
+        # Split sentence on EOS and Added-Tokens
+        sent_list = re.split(pattern, sentence)
+        # remove empty elements
+        sent_list = [item for item in sent_list if item]
+
         # Tokenize each sentence separately.
         segmented = []
         for _sentence in sent_list:
-            # Locate the mask-before placeholders
-            # (to zero-out the prompt loss during LM finetuning).
-            _sentence_chunks = _sentence.split(DefaultTokens.MASK_BEFORE)
-            # Tokenize each chunk separately and insert the padding token.
-            # between each sequence of tokens.
-            _sentence_tokens = []
-            for _chunk in _sentence_chunks:
-                _sentence_tokens += self.tokenize_string(_chunk, side, is_train) + [
-                    self.pad_token  # not sure this covers all cases
-                ]
-            # Re-insert the eos token.
-            segmented += _sentence_tokens[:-1] + [self.eos_token]
-        return segmented[:-1]
+            if _sentence in delim_list:
+                segmented.append(_sentence)
+            else:
+                # Locate the mask-before placeholders
+                # (to zero-out the prompt loss during LM finetuning).
+                _sentence_chunks = _sentence.split(DefaultTokens.MASK_BEFORE)
+                # Tokenize each chunk separately and insert the padding token.
+                # between each sequence of tokens.
+                _sentence_tokens = []
+                for _chunk in _sentence_chunks:
+                    _sentence_tokens += self.tokenize_string(_chunk, side, is_train)
+                    if _chunk[-1] == " ":
+                        trailingspace = self.tokenize_string(" a", side, is_train)[0][
+                            :-1
+                        ]
+                        _sentence_tokens += [trailingspace]
+                    _sentence_tokens += [
+                        self.pad_token  # not sure this covers all cases
+                    ]
+                segmented.extend(_sentence_tokens[:-1])
```

Comment:
This logic is probably broken. Maybe can just call it before applying the new mapped_tokens based logic?

Reply:
removing this for now.